File: Data Structures & Algorithms/find-pivot-index/submission-0.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        # [1] left = 0 right = 29
        # [7] left = 1 right = 20
        # [3] left = 8 right = 17
        # [6] left = 11 right 11

        left = []
        total = 0
        for n in nums:
            left.append(total)
            total += n
        
        total = 0
        right = [0] * len(nums)
        for i in range(len(nums) - 1, -1, -1):
            right[i] = total
            total += nums[i]
        
        for i in range(len(nums)):
            if right[i] == left[i]:
                return i;
        
        return -1
            

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public List<int> RightSideView(TreeNode root) {
        var output = new List<int>();
        var queue = new Queue<TreeNode>();

        if (root != null)
            queue.Enqueue(root);

        while (queue.Count > 0) {
            var currentLevelLength = queue.Count;
            var currentNodePos = 0;
            for (int i = 0; i < currentLevelLength; i++) {
                var node = queue.Dequeue();

                if (currentLevelLength > 1) {
                    if (currentNodePos == currentLevelLength - 1)
                        output.Add(node.val);
                    else
                        currentNodePos++;
                } else
                    output.Add(node.val);

                if (node.left != null)
                    queue.Enqueue(node.left);
                if (node.right != null)
                    queue.Enqueue(node.right);
            }
        }

        return output;
    }
}
